jsettlers.common/src/main/resources/jsettlers/common/buildings/amazon/baker.xml

```diff
@@ -133,11 +133,11 @@
 
 	<job name="C_work1" type="LOOK_AT"       successjob="C_work2" 	failjob="gohome" direction="NORTH_EAST"/>
 	<job name="C_work2" type="PLAY_ACTION1"  successjob="C_work3" 	failjob="gohome" time="1"/>
-	<job name="C_work3" type="SET_MATERIAL"  successjob="C_work4"  	failjob="gohome" material="BLADE"/>
-	<job name="C_work4" type="SMOKE_ON"      successjob="C_work5"	failjob="gohome" dx="0" dy="2"/>
-	<job name="C_work5" type="WAIT"          successjob="C_work6" 	failjob="gohome" time="4"/>
-	<job name="C_work6" type="SET_MATERIAL"  successjob="C_work7"  	failjob="gohome" material="BREAD"/>
-	<job name="C_work7" type="SMOKE_OFF"     successjob="C_work8" 	failjob="gohome" dx="0" dy="2"/>
+	<job name="C_work3" type="SET_MATERIAL"  successjob="smoke"  	failjob="gohome" material="BLADE"/>
+
+	<job name="smoke" 	type="BUILDING_ANIMATION"	successjob="C_work6" 	failjob="gohome" takeMaterialFromMap="false" />
```

Comment:
This kind of abuses takeMaterialFromMap.
You should introduce a new boolean variable that actually describes what it is doing.